flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -157,14 +203,25 @@ public boolean reachedEnd() throws IOException {
 	 * @throws java.io.IOException
 	 */
 	@Override
-	public OUT nextRecord(OUT tuple) throws IOException {
+	public Row nextRecord(Row row) throws IOException {
 		try {
-			resultSet.next();
-			if (columnTypes == null) {
-				extractTypes(tuple);
+			hasNext = resultSet.next();
+			if(!hasNext) {
+				return null;
+			}
+			try{
+				//This throws a NPE when the TypeInfo is not passed to the InputFormat,
+				//i.e. KryoSerializer used to generate the passed row
+				row.productArity();
```

Comment:
The problem is that the Row object is instantiated by the DataSourceTask and, if no type info is provided, it's created by the KryoSerializer and row.fields in that case is null and there's no way to instantiate it (or I don't know how to do that). Is there a way to initialize them?